crates/uv-cache/src/lib.rs

```diff
@@ -355,10 +356,20 @@ impl Cache {
                 if bucket.is_dir() {
                     for entry in walkdir::WalkDir::new(bucket) {
                         let entry = entry?;
-                        if entry.file_type().is_symlink() {
-                            if let Ok(target) = fs_err::canonicalize(entry.path()) {
-                                references.insert(target);
-                            }
+
+                        // Identify entries that match the wheel stem pattern (e.g., `typing-extensions-4.8.0-py3-none-any`).
+                        let Some(filename) = entry
+                            .path()
+                            .file_name()
+                            .and_then(|file_name| file_name.to_str())
+                        else {
+                            continue;
+                        };
+                        if WheelFilename::from_stem(filename).is_err() {
+                            continue;
+                        }
+                        if let Ok(target) = uv_fs::resolve_symlink(entry.path()) {
```

Comment:
And here.

Although I guess we weren't emitting logs before. I wonder if it might help debugging if something goes wrong here.